jwk.go

```diff
@@ -0,0 +1,57 @@
+package clerk
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/go-jose/go-jose/v3"
+)
+
+type JSONWebKeySet struct {
+	APIResource
+	Keys []JSONWebKey `json:"keys"`
+}
+
+type JSONWebKey struct {
+	APIResource
+	jose.JSONWebKey
+	Key       any    `json:"key"`
+	KeyID     string `json:"kid"`
+	Algorithm string `json:"alg"`
+	Use       string `json:"use"`
+}
+
+func (k *JSONWebKey) UnmarshalJSON(data []byte) error {
```

Comment:
I wanted to avoid using a go-jose type and instead make sure we use only our own package's (clerk) types everywhere.

There's no reason to expose types that are meant for internal use anywhere. We unfortunately do this in v1 of our SDK.

The go-jose library is able to decode JSON web keys and we can leverage that to get a response from our Backend API GET /v1/jwks endpoint and translate it to our own `clerk.JSONWebKeySet` and effectively `clerk.JSONWebKey`.

Unfortunately, it looks like extracting the key ID and key from a JWK response is not that straightforward. Depending on the algorithm that's used the JWK response might have different fields set, each with different meaning.

The go-jose library [handles unmarshaling nicely](https://github.com/go-jose/go-jose/blob/47edce0854d533ac27795c9befd90b1f7ef87554/jwk.go#L166), but its [raw unmarshaled type is unexported](https://github.com/go-jose/go-jose/blob/47edce0854d533ac27795c9befd90b1f7ef87554/jwk.go#L42). 

So, what I thought we could do is use the jose.JSONWebKey type to unmarshal the raw JSON and then copy over the resulting fields to our type.

I initially tried type aliasing but it didn't work as expected.

Hope the above makes the decision clearer. Let me know if I should offer more details! Perhaps I can add a comment explaining what's going on here?

Reply:
> I wanted to avoid using a go-jose type and instead make sure we use only our own package's (clerk) types everywhere. There's no reason to expose types that are meant for internal use anywhere. We unfortunately [do this](https://github.com/clerk/clerk-sdk-go/blob/8bd66b16f28be05ad4377d3f24d2ace2c65d5fb3/clerk/tokens.go#L43) in v1 of our SDK.

Very good call. This provides us with flexibility to change the underlying library (go-jose) without it being a breaking change.

Reply:
I see, yes good call indeed. Now that the field is unexported the purpose is also much clearer!